stylecheck/lint.go

```diff
@@ -80,6 +80,30 @@ func CheckDotImports(pass *analysis.Pass) (interface{}, error) {
 	return nil, nil
 }
 
+func CheckDuplicatedImports(pass *analysis.Pass) (interface{}, error) {
+	for _, f := range pass.Files {
+		// Collect all imports by their import path
+		imports := make(map[string][]*ast.ImportSpec, len(f.Imports))
+		for _, imp := range f.Imports {
+			if current, ok := imports[imp.Path.Value]; !ok {
```

Comment:
This entire if/else can be replaced with an unconditional `imports[imp.Path.Value] = append(imports[imp.Path.Value], imp)`